notebooks/work-in-progress/fuel-price-imputation/fuel-price-imputation-zane.ipynb

```diff
@@ -0,0 +1,1133 @@
+{
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #14.        ("hist_gbr", HistGradientBoostingRegressor(loss="absolute_error", categorical_features=cat_cols_idx)),
</pre><p>The default value of <code>max_iter</code> is 100, which is probably too low (also why the best learning rate is pretty high - the model is trying to make the most of a small number of iterations). We can probably crank that up to like 1000 and rely on early stopping.</p><p>BUT this will lead to much longer training times, and as you've seen the output is not that sensitive to this stuff. So for the purposes of experimentation, I think it's good to leave as is, but at the very end of this whole process let's do one big round of grid searching to dial it in.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/catalyst-cooperative/pudl/pull/1696/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Reply:
<p>I've switched over to using a single set of parameters for basic experimentation, and it seems to train in a reasonable amount of time, even with all these columns. But yeah I agree a big parameters space search at the end seems reasonable!</p><p> </p><p>I'll mess around with some of the other model params in a single-shot and see how they behave. Do you have an intuitive sense of what the minimum number of samples per leaf does? Does it prevent overfitting on smaller datasets because the model can only get so granular?</p>

Reply:
Yep, `min_samples_leaf` helps prevent overfitting aka fitting to noise in the training data. A groupby slices your data up into equally spaced bins and then calculates an aggregate of the points in each bin. A decision tree also slices your data up into bins (called "leaves") and aggregates the points, but instead of equal spacing the sizes/shapes of the bins are calculated dynamically. `min_samples_leaf` sets the minimum allowable sample size in the bins so they don't get overly fit on a small, potentially noisy sample. If you had only 3 points in a bin, you can imagine the aggregate could get thrown way off by any noise, whereas a 30 point aggregate would be less bothered.